src/rgw/motr/sync/motr_sync_impl.cc

```diff
@@ -29,46 +47,65 @@ int MotrLock::lock(const std::string& lock_name,
   }
   int rc = 0;
   motr_lock_info_t lock_obj;
-  rc = _lock_provider->read_lock(lock_name, &lock_obj);
-  if (rc != 0 && rc != -ENOENT)
-    return rc;
+  // First, try to write lock object
+  struct motr_locker_info_t locker;
+  std::string s_locker_id = locker_id;
+  if (s_locker_id.empty()) {
+    s_locker_id = random_string(32);
+  }
+  locker.cookie = s_locker_id;
+  locker.expiration = ceph_clock_now() + lock_duration;
+  locker.description = "";
+  // Insert lock entry
+  lock_obj.lockers.insert(
+      std::pair<std::string, struct motr_locker_info_t>(locker_id,
+                                                        locker));
+  rc = _lock_provider->write_lock(lock_name, &lock_obj, false);
   if (rc == 0) {
-    // lock entry is present. Check if this is a stale/expired lock
-    utime_t now = ceph_clock_now();
-    auto iter = lock_obj.lockers.begin();
-    while (iter != lock_obj.lockers.end()) {
-      struct motr_locker_info_t &info = iter->second;
-      if (!info.expiration.is_zero() && info.expiration < now) {
-        // locker has expired; delete it
-        iter = lock_obj.lockers.erase(iter);
-      } else {
-        ++iter;
+    // Possiblity that this lock, after creation here, is deleted by
+    // another client. So, try to get lock object (after some millisecs)
+    // to see if it still exists - reconfirm lock.
+    sleep(RECONFIRM_LOCK_TIME_DELAY);
+    motr_lock_info_t cnfm_lock_obj;
+    rc = _lock_provider->read_lock(lock_name, &cnfm_lock_obj);
```

Comment:
Not very sure whether we should  confirm that lock is indeed acquired

Reply:
@sachinpunadikar : This was what proposed by @sjain09. This is to consider race condition mentioned by him.